server/src/main.rs

```diff
@@ -1,6 +1,196 @@
+        .form(&params)
+        .send()
+        .await
+    {
+        match res.json::<TokenResponse>().await {
+            Ok(token) => {
+                let res = client
+                    .post("https://api.github.com/graphql")
+                    .header("authorization", format!("bearer {}", &token.access_token))
+                    .body(include_str!("query.json"))
+                    .send()
+                    .await;
+
+                if let Ok(res) = res {
+                    use structs::GraphQLQuery;
+
+                    let query = res.json::<GraphQLQuery>().await.unwrap();
+
+                    let pods: Vec<_> = query
+                        .data
+                        .viewer
+                        .organization
+                        .team
+                        .childTeams
+                        .nodes
+                        .iter()
+                        .filter(|p| {
+                            p.members
+                                .nodes
+                                .iter()
+                                .any(|m| m.login == query.data.viewer.login)
+                        })
+                        .collect();
+
+                    if pods.is_empty() {
+                        let response = Response::builder()
+                            .status(424)
+                            .body("error: failed dependency (pods)");
+
+                        return Ok(response);
+                    }
+
+                    let pod_id: Result<i32, diesel::result::Error> =
+                        match diesel::insert_into(data::pod::dsl::pod)
+                            .values(data::NewPod {
+                                pod_id: pods[0].name.clone(),
+                                pod_name: pods[0].description.clone(),
+                            })
+                            .returning(data::pod::id)
+                            .get_result::<i32>(&d.get().unwrap())
+                        {
+                            Ok(r) => Ok(r),
+                            Err(_) => {
+                                let r = match data::pod::dsl::pod
+                                    .filter(data::pod::pod_id.eq(pods[0].name.clone()))
+                                    .load::<(i32, String, String)>(&d.get().unwrap())
+                                {
+                                    Ok(r) => Ok(r[0].0),
+                                    Err(e) => Err(e),
+                                };
+
+                                r
+                            }
+                        };
+
+                    if pod_id.is_err() {
+                        let response = Response::builder()
+                            .status(424)
+                            .body("error: failed dependency (pod_id)");
+
+                        return Ok(response);
+                    }
+
+                    let _ = diesel::insert_into(data::author::dsl::author)
```

Comment:
Anything bound to `_` is immediately dropped so I don't think this is needed.
```suggestion
                    diesel::insert_into(data::author::dsl::author)
```

Reply:
The `_` prevents a `Unhandled Result` warning

Reply:
That's interesting – not sure why! Maybe something to investigate after the hackathon.